rand_chacha/src/lib.rs

```diff
@@ -16,10 +16,24 @@
 #![deny(missing_debug_implementations)]
 #![doc(test(attr(allow(unused_variables), deny(warnings))))]
 
-#![no_std]
+#![cfg_attr(not(feature = "std"), no_std)]
 
+extern crate c2_chacha;
+pub extern crate generic_array;
 pub extern crate rand_core;
 
 mod chacha;
 
-pub use chacha::{ChaChaRng, ChaChaCore};
+use generic_array::typenum;
+
+pub use chacha::{ChaChaXRng, ChaChaXCore};
```

Comment:
I think it might be better to be more conservative and not expose RNGs that have a non-default number of rounds. So maybe this should not be public.

Reply:
Yeah, that makes sense.

I'll make opaque wrappers for the supported rngs to avoid exposing the now-private parameterizable type, so they'll all be first-class items to Rustdoc. This will also remove generic-array from the interface, so I'll see if I can do the implementation without it